File: ipreneur/frontend/src/services/api/deckTypes.ts
import { apiClient } from "@/services/api/client";

/** One type-specific intake field, as declared by the backend registry. */
export interface BriefField {
  name: string;
  label: string;
  type: "text" | "textarea" | "number";
  placeholder: string;
  help: string;
  required: boolean;
}

export interface DeckTypeOption {
  key: string;
  label: string;
  description: string;
  eyebrow: string;
  slideCount: number;
  research: "full" | "light" | "none";
  supportsStages: boolean;
  /** Shared form fields this type needs when there is no website to crawl. */
  manualRequired: string[];
  briefFields: BriefField[];
}

/**
 * The deck-type catalogue, served from the backend registry.
 *
 * The new-project form renders its picker and its per-type fields from this
 * rather than a local copy, so the form cannot ask for fields the prompt does
 * not read (or miss ones it requires).
 */
export const deckTypesApi = {
  list: async (): Promise<DeckTypeOption[]> => {
    const { data } = await apiClient.get("/deck-types");
    return data.data;
  },
};

File: ipreneur/backend/app/api/v1/endpoints/deck_types.py
"""
Deck-type catalogue endpoint.

The new-project form renders its deck-type picker and its per-type fields from
this response rather than from a hard-coded copy of the registry, so the form
cannot drift from what the generation prompt actually consumes.
"""
from fastapi import APIRouter

from app.decks.registry import serialize_deck_types

router = APIRouter()


@router.get("")
async def list_deck_types() -> dict:
    """Every deck type: label, description, slide count, and its intake fields."""
    return {"data": serialize_deck_types()}
